### PR TITLE
feat: new tx filter format

### PR DESCRIPTION
**Pull request by ratik**
now cosmos relayer supports new format for [tx filter](https://p2pvalidator.atlassian.net/jira/software/projects/LSC/boards/121?selectedIssue=LSC-121) so the interchain queries contract was updated
Testing tips: 
1. neutron from master
2. cosmos-query-relayer from [feature/new-transaction-filter](https://github.com/neutron-org/cosmos-query-relayer/pull/26)

### CODE REVIEW

File: packages/interchain_queries/src/register_queries.rs [hunk 1]
```diff
@@ -164,8 +164,11 @@ pub fn register_transfers_query(
     recipient: String,
     update_period: u64,
 ) -> ContractResult<Response<NeutronMsg>> {
-    let query_data = TransferRecipientQuery { recipient };
-
+    let query_data: Vec<TransactionFilterItem> = vec![TransactionFilterItem {
+        field: "transfer.recipient".to_string(),
```
Comment (pr0n00gler) [inline packages/interchain_queries/src/register_queries.rs:168]: Define a constant, please

File: packages/interchain_queries/src/sudo.rs [hunk 2]
```diff
@@ -80,8 +106,17 @@ pub fn sudo_tx_query_result(
 
                 // Parse a Send message and check that it has the required recipient.
                 let transfer_msg: MsgSend = MsgSend::decode(message.value.as_slice())?;
-
-                if transfer_msg.to_address == query_data.recipient {
+                let recipient = query_data.iter().fold("", |prev, row| {
+                    if row.field == "transfer.address" && row.op.as_str() == "eq" {
```
Comment (pr0n00gler) [inline packages/interchain_queries/src/sudo.rs:110]: `"transfer.address"`? Don't we need a `"transfer.recipient"` here?

Comment (swelf19) [inline packages/interchain_queries/src/sudo.rs:110]: You can derive PartialEq and Eq traits with a `TransactionFilterOp` definition and compare row.op with a TransactionFilterOp::Eq not a str

File: packages/interchain_queries/src/sudo.rs [hunk 3]
```diff
@@ -80,8 +106,17 @@ pub fn sudo_tx_query_result(
 
                 // Parse a Send message and check that it has the required recipient.
                 let transfer_msg: MsgSend = MsgSend::decode(message.value.as_slice())?;
-
-                if transfer_msg.to_address == query_data.recipient {
+                let recipient = query_data.iter().fold("", |prev, row| {
```
Comment (swelf19) [inline packages/interchain_queries/src/sudo.rs:109]: don't you think find(+map) looks simpler at this place?

**Comment by ratik** [comment]
sorry guys, had to rebase it bc some refactoring merged before this PR

**Review by swelf19 — APPROVED** [review]

**Review by sotnikov-s — CHANGES_REQUESTED** [review]
I think it would be very useful if you added the complicated query support to the neutron_interchain_queries contract, cuz now it only accepts the recipient and inside constructs a simple query `transfer.recipient == recipient`. here's why:
1. I'd like to test it with a complicated query, but currently it's impossible;
2. in integration tests it is also a crucial thing.

**Comment by ratik** [comment]
> I think it would be very useful if you added the complicated query support to the neutron_interchain_queries contract, cuz now it only accepts the recipient and inside constructs a simple query `transfer.recipient == recipient`. here's why:
> 
> 1. I'd like to test it with a complicated query, but currently it's impossible;
> 2. in integration tests it is also a crucial thing.
> 
> so please add this functionality and update test_tx_query_result.sh accordingly

agreed with @sotnikov-s that we need add one more RegisterTransfersQuery but with filtering by height

**Review by oopcode — APPROVED** [review]
tested with `test_tx_query_result.sh`

File: packages/interchain_queries/src/types.rs [hunk 4]
```diff
@@ -49,6 +50,64 @@ pub const PARAMS_STORE_KEY: &str = "params";
...
+        if let Some(n) = n {
+            Ok(Self::Int(n.into()))
+        } else {
+            let n = v
+                .as_str()
+                .ok_or_else(|| D::Error::custom("Value must be number or string"))?;
+            Ok(Self::String(n.to_string()))
+        }
+    }
+}
```
Comment (ratik) [inline packages/interchain_queries/src/types.rs:90]: Please pay attention here

Comment (ratik) [reply packages/interchain_queries/src/types.rs:90]: @oopcode

**Review by oopcode — APPROVED** [review]
tested with `test_tx_query_result.sh`, everything seems to work fine now

**Review by sotnikov-s — APPROVED** [review]

**Review by oopcode — APPROVED** [review]

**Pull request merged**